### ISSUE TITLE
set-nth and arglists don't play well together

### ISSUE THREAD

**Issue by HugoGiraudel** [sass/libsass#1255]
## Test

```scss
@function double($value) {
  @return $value * 2;
}

@mixin dummy-bug($args...) {
  @for $i from 1 through length($args) {
    $args: set-nth($args, $i, double(nth($args, $i)));
  }
  
  content: $args;
}

.foo {
  @include dummy(1, 2, 3, 4);
}
```

## Expected

```css
.foo {
  content: 2, 4, 6, 8;
}
```

## Result (LibSass 3.2.x)

Compile never ends.

## Workaround

```scss
@mixin dummy($args...) {
  $list: ();
  
  @each $arg in $args {
    $list: append($list, double($arg), 'comma');
  }
  
  content: $list;
}
```

## Observations

As far as I can tell, there is somehow an issue when trying to use `set-nth` on an `arglist`. Maybe is it only restricted when using a loop? I must say I don't know.

**Comment by xzyfer** [comment]
This may have the same underlying cause as https://github.com/sass/libsass/issues/1171

**Comment by xzyfer** [comment]
Fixed by #1318.
